board/app/models/thread.php

```diff
@@ -56,67 +83,97 @@ public static function getAll($offset, $limit)
         $threads = array();
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
-
-        foreach($rows as $row) 
-        {
+            
+        foreach($rows as $row) {
             $threads[] = new self($row);
         }
-
-    return $threads;
+        return $threads;
     }
 
-    public static function countAll()
+    public static function countAll() 
     {
         $db = DB::conn();
         return (int) $db->value('SELECT COUNT(*) FROM thread');
     }
 
-    public function countComments()
+    public static function countAllThreadByUserId($user_id) 
     {
         $db = DB::conn();
-        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ? ", array($this->id));
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($user_id));
     }
 
+    public static function countAllThreadByCategory($category) 
+    {
+        $db = DB::conn();
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($category));
+    }
 
-    public static function get($id)
+    public function countComments() 
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ? ", array($this->id));
```

Comment:
Same po with the previous comment, this function is unused and i already deleted this po. :bow: thank you po.
-->https://github.com/maragatchalian/FoodFacts/commit/a02c673044a5554a4682b00868e87bf88df6ea55